Change CommandBuffer::pushConstants api to match other APIs

playmer opened the pull request:
Most of the APIs in VkHLF expect shared_ptrs of the types VkHLF creates, this matches that. This could also go some other ways, and I'm willing to make changes in that case. Perhaps we take a reference to a shared_ptr since we don't actually need the shared_ptr. We could also make this an overload so we maintain the previous API as well. Happy to make any changes.

playmer committed:
Fix

asuessenbach commented:
Thanks for catching that glitch.
Just two little notes on it:
1. We would prefer to have a std::shared_ptr<...> const& layout as the argument;
2. You don't need to explicitly cast that shared_ptr to an vk::PipelineLayout. Just dereferencing would do the job.
If you would adjust your pull request accordingly, we're happy to accept it.

playmer commented:
Of course, happy to do it. I'm away from my desktop, but I'll make the requested changes tomorrow and push them up. Thanks for the feedback.

playmer committed:
Make requested changes

- No longer explicitly cast to vkPipelineLayoit
- Changed from `std::shared_ptr<...>` to `std::shared_ptr<...> const &`
